src/server/game/Battlegrounds/Battleground.cpp

```diff
@@ -1076,6 +1076,12 @@ void Battleground::StartBattleground()
         TC_LOG_DEBUG("bg.arena", "Arena match type: %u for Team1Id: %u - Team2Id: %u started.", m_ArenaType, m_ArenaTeamIds[TEAM_ALLIANCE], m_ArenaTeamIds[TEAM_HORDE]);
 }
 
+void Battleground::TeleportPlayerToExploitLocation(Player* player)
+{
+    if (WorldSafeLocsEntry const* loc = GetExploitTeleportLocation(player->GetTeamId()))
```

Comment:
one thing that just came to my mind - shouldn't GetBGTeam be used here if this detection also applies to arenas (are there worldsafelocs for that?)